### PR TITLE
Mark Label Chunks Test or Train

### PR DESCRIPTION
**Pull request by waltaskew**
- Add an 'in_test_set' field to our label chunks.
- Mark each chunk as in the test set or not.

### CODE REVIEW

File: usl_pipeline/usl_lib/tests/storage/test_metastore.py [hunk 1]
```diff
@@ -211,3 +211,77 @@ def test_flood_scenario_config_delete():
...
+        cell_size=3.0,
+        crs="over there",
+        chunk_size=1,
+        chunk_x_count=2,
+        chunk_y_count=5,
+    )
+
+    # Calculate whether all chunks are in the validation set.
+    chunks_in_validation_set = [
+        metastore.SimulationLabelChunk.is_in_validation_set(study_area, "config", 0, 0),
```
Comment (rsutormin) [inline usl_pipeline/usl_lib/tests/storage/test_metastore.py:233]: Walt, can you also show in test code that several consequent calls to the same parameters return the same result (even though there is random generator involved). I know there is a seed initiated by hasher. But still it's good to double check that everything behaves as expected.

**Review by rsutormin — APPROVED** [review]

File: usl_pipeline/usl_lib/tests/storage/test_metastore.py (hunk 1 above)
Comment (waltaskew) [reply usl_pipeline/usl_lib/tests/storage/test_metastore.py:233]: Sure, I can do that, although we're essentially doing that because if not for a seed, the tests would be very flaky! :)

**Comment by waltaskew** [comment]
@prashantkul @tatianawu can y'all confirm if this matches the test & validate behaviour y'all are expecting? We'll need to merge this and re-run data pipelines to get these changes out.

**Comment by prashantkul** [comment]
Maybe I didn't understand correctly - are you creating a test/train set or
a validation/train set? They're different and ideally we want all 3 of
them. Training set needs to be randomly shuffled.

**Comment by waltaskew** [comment]
> Maybe I didn't understand correctly - are you creating a test/train set or a validation/train set? They're different and ideally we want all 3 of them. Training set needs to be randomly shuffled.

Sorry, train & validate. When generating the labels, we mark them as either part of train or validate in the data pipeline.
The test & train will need to happen later in the dataset generation.

**Comment by tatianawu** [comment]
> > Maybe I didn't understand correctly - are you creating a test/train set or a validation/train set? They're different and ideally we want all 3 of them. Training set needs to be randomly shuffled.
> 
> Sorry, train & validate. When generating the labels, we mark them as either part of train or validate set in the data pipeline. The test & train will need to happen later in the dataset generation.

Can you explain what you mean by the train/test split needs to happen later?

We need a train/test split. We don't technically need a separate validation set; we can choose to randomly split off part of each batch for validation. I actually think it would be fine to just switch the current validation set you're creating to being the test set.

**Comment by prashantkul** [comment]
I think we want a Train/Test first. Validation can be split off from the
training set. Looks like that was the intention of the PR based on its
title.

**Comment by prashantkul** [comment]
yes, I agree.

**Comment by waltaskew** [comment]
Sorry, I may have my terminology mixed up.
What I'm doing here is marking 20% of labels as 'in_validation_set'
Should I have called it 'in_test_set' instead?

I thought y'all wanted the data pipeline to mark 20% of the labels as something like that?

**Comment by prashantkul** [comment]
ah, it's a terminology issue. Yes it should have been "in_test_set"



*Prashant Kulkarni*

Customer Engineer, Security & Compliance Specialist

Google Cloud

**Comment by waltaskew** [comment]
Okay! Sorry for the confusion! The branch name is `simulation-test-train-indicators` which proves I chose the word 'test' and 'validate' at random.

I've updated the flag to 'in_test_set'

File: usl_pipeline/cloud_functions/main_test.py [hunk 2]
```diff
@@ -1228,6 +1241,7 @@ def test_collapse_city_cat_output_chunks(mock_firestore_client):
                     ),
                     "x_index": 0,
                     "y_index": 1,
+                    "in_validation_set": False,
```
Comment (tatianawu) [inline usl_pipeline/cloud_functions/main_test.py:1244]: Your tests still use `in_validation_set` in some places.

Comment (waltaskew) [reply usl_pipeline/cloud_functions/main_test.py]: yep, I've now fixed this and the build is happy

File: usl_pipeline/usl_lib/tests/storage/test_metastore.py [hunk 3]
```diff
@@ -211,3 +211,99 @@ def test_flood_scenario_config_delete():
...
+        x_ll_corner=1.0,
+        y_ll_corner=2.0,
+        cell_size=3.0,
+        crs="over there",
+        chunk_size=1,
+        chunk_x_count=2,
+        chunk_y_count=5,
+    )
+
+    # Calculate whether all chunks are in the validation set.
```
Comment (tatianawu) [inline usl_pipeline/usl_lib/tests/storage/test_metastore.py:231]: Nit: Some of the comments still refer to this as the validation set, which might cause some confusion.

**Review by tatianawu — APPROVED** [review]

File: usl_pipeline/usl_lib/tests/storage/test_metastore.py (hunk 3 above)
Comment (waltaskew) [reply usl_pipeline/usl_lib/tests/storage/test_metastore.py]: sorry, removed those too
